clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -905,6 +971,57 @@ public void testSendTxnOffsetsWithGroupId() {
         }
     }
 
+    private double getAndAssertDuration(KafkaProducer<?, ?> producer, String name, double floor) {
+        double value = getMetricValue(producer, name);
+        assertTrue(value > floor);
+        return value;
+    }
+
+    @Test
+    public void testMeasureTransactionDurations() {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, "some.id");
+        configs.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, 10000);
+        configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9000");
+        Time time = new MockTime(1);
+        MetadataResponse initialUpdateResponse = RequestTestUtils.metadataUpdateWith(1, singletonMap("topic", 1));
+        ProducerMetadata metadata = newMetadata(0, Long.MAX_VALUE);
+
+        MockClient client = new MockClient(time, metadata);
+        client.updateMetadata(initialUpdateResponse);
+        client.prepareResponse(FindCoordinatorResponse.prepareResponse(Errors.NONE, "some.id", host1));
+        client.prepareResponse(initProducerIdResponse(1L, (short) 5, Errors.NONE));
+
+        try (KafkaProducer<String, String> producer = kafkaProducer(configs, new StringSerializer(),
+            new StringSerializer(), metadata, client, null, time)) {
+            producer.initTransactions();
+            assertTrue(getMetricValue(producer, "txn-init-time-total") > 999999);
```

Comment:
I am not sure I understand this verification. Could you elaborate?

Reply:
I'm verifying that something was measured and that it's at least 1 tick of the clock. The clock is shared between multiple threads (e.g. the io threads) so the number of ticks depends  on what threads get scheduled while we're in `initTransactions`.

Reply:
Got it!
Could you please put the `999999` in a variable and give it a meaningful name such as `lowerBoundNs`. Maybe put also the `1` that you use in the `MockTime` constructor in a variable such as `tickMs` so that the difference in units becomes clear. 

BTW: you could also use `getAndAssertDuration()` here, right?

Reply:
I now saw that in the consumer tests you use `Duration.ofSeconds(1).toMillis()` and `Duration.ofMillis(999).toNanos()`. This makes it already clearer. I think a variable with a meaningful name for the lower bound would make it even clearer.